src/v/cluster/feature_barrier.cc

```diff
@@ -161,7 +161,8 @@ feature_barrier_response feature_barrier_state_base::update_barrier(
       entered);
     auto i = _barrier_state.find(tag);
     if (i == _barrier_state.end()) {
-        _barrier_state[tag] = feature_barrier_tag_state({{peer, entered}});
```

Comment:
unrelated: for a constructor `foo(std::pair)` it seems that `foo({..})` and `foo({{..}})` bot work. is one of those implicitly creating an initializer list?

Reply:
I'm pretty sure that's not a context where an initializer list could be generated. It could just be that the outer braces represent `feature_barrier_tag_state`, sort of equivalent to calling the copy-constructor.

Reply:
ahh yeh right, that makes sense